pkgs/tools/package-management/nix/default.nix

```diff
@@ -182,6 +182,20 @@ in lib.makeExtensible (self: {
     sha256 = "sha256-hNHfvmb1bIWwqFT5nesQgwh4V0OlyZHxj5ZVSQbZ+p4=";
   };
 
+  # The minimum Nix version supported by Nixpkgs
+  minimum =
+    let
+      minver = import ../../../../lib/minver.nix;
+      major = lib.versions.major minver;
+      minor = lib.versions.minor minver;
+      attribute = "nix_${major}_${minor}";
+      nix = self.${attribute};
+    in
+    if ! self ? ${attribute} then
+      throw "The minimum supported Nix version is ${minver} (declared in lib/minver.nix), but pkgs.nixVersions.${attribute} does not exist."
+    else
+      nix;
```

Comment:
```suggestion
    if self ? ${attribute} then
      nix
    else
      throw "The minimum supported Nix version is ${minver} (declared in lib/minver.nix), but pkgs.nixVersions.${attribute} does not exist.";
```

Reply:
I think for code that performs validation, it's good for the error case to be in the `then`, even if that leads to a negation.